### PR TITLE
`Taxis example, update docs and deps

### PR DESCRIPTION
**Pull request by dberardo-com**
taxi example wont start due these errors:

poetry install --> ModuleNotFoundError: No module named 'platformdirs'
poetry install --> ModuleNotFoundError: No module named 'pandas'

Also, added some more user-friendly explanation on the readme.

### CODE REVIEW

**Comment by dberardo-com** [comment]
@MaxHalford i assume that the train.py script is optional, but one could still use it to "warm up" the model before starting the simulation. is this right?

File: docker-compose.yaml [hunk 1]
```diff
@@ -1,6 +1,7 @@
 version: '3'
 services:
   app:
+    container_name: app
```
Comment (MaxHalford) [inline docker-compose.yaml:4]: What benefit does this provide? I think setting a `container_name` becomes annoying when you want to deploy multiple copies of the service

File: examples/taxis/README.md [hunk 2]
```diff
@@ -2,17 +2,41 @@
 
 Simulating real traffic using the [*New York City Taxi Trip Duration* dataset](https://www.kaggle.com/c/nyc-taxi-trip-duration).
 
+## General idea
+
+This proccedure helps understand how beaver can be used to predict trip durations of taxis using an event dataset consisting of pickup date-times, pickup coordinates, arrival coordinates, etc.
```
Comment (MaxHalford) [inline examples/taxis/README.md]:
```suggestion
This procedure helps understand how Beaver can be used to predict taxi trip durations using an event dataset. Each event contains a pickup timestamp, pickup coordinates, arrival coordinates, etc.
```

File: examples/taxis/README.md [hunk 3]
```diff
@@ -2,17 +2,41 @@
...
+
+* The procedure here is to create an empty model using the river library
```
Comment (MaxHalford) [inline examples/taxis/README.md]:
```suggestion
* The procedure here is to create an empty model using River.
```

File: examples/taxis/README.md [hunk 4]
```diff
@@ -2,17 +2,41 @@
...
+* Serialize the model using dill
+* Upload the model using the beaver http client --> you could also do this using the HTTP API directly (see localhost:3000/docs for more details)
```
Comment (MaxHalford) [inline examples/taxis/README.md]:
```suggestion
* Upload the model using Beaver's HTTP client -- you could also do this using the HTTP API directly (see `localhost:3000/docs` for more details)
```

File: examples/taxis/README.md [hunk 5]
```diff
@@ -2,17 +2,41 @@
...
+* Then start receiving trip data in a simulation and make predictions for trip duration
+
+During simulation you should see the error decreasing with time, which can be visualized in the output messages as in:
```
Comment (MaxHalford) [inline examples/taxis/README.md]:
```suggestion
During the simulation you should see the error decreasing with time, which can be visualized in the output messages, as so:
```

File: examples/taxis/README.md [hunk 6]
```diff
@@ -2,17 +2,41 @@
...
+
+```
+... a taxi starts a trip with id 000014
+#0000014 departs at 2016-01-01 00:04:57
+... times goes by
+... eventually the trip ends
+#0000014 arrives at 2016-01-01 00:08:37, took 0:03:40, predicted 0:08:41
+```
+
+so for the trip 000014 we can see an error of more then 200% in this case
```
Comment (MaxHalford) [inline examples/taxis/README.md]:
```suggestion
For this trip number 000014, we can see an error of over 200%.
```

File: examples/taxis/README.md [hunk 7]
```diff
@@ -2,17 +2,41 @@
...
+
 ## Setup
 
 First of all you need a running instance of Beaver. For instance, you can to the root of this repo and start an instance in the background:
 
 ```sh
-(cd ... && docker-compose up -d)
+(cd ... && docker-compose up -d && docker exec -it app /bin/bash)
```
Comment (MaxHalford) [inline examples/taxis/README.md:31]: Actually the idea is to run the simulation from the host, not from the instance where Beaver is located.

File: docker-compose.yaml (hunk 1 above)
Comment (dberardo-com) [reply docker-compose.yaml:4]: the benefit is that the documentation is consistent with the environment of the user. i think users will never need to run multiple instances of the test container anyway. i find it a good practice, if you dont like it, then i can build in a filter function to get the container id.

File: examples/taxis/README.md (hunk 7 above)
Comment (dberardo-com) [reply examples/taxis/README.md:31]: i see, but i find it nicer not to have to install anything on my dev machine. If you like it that way i could setup an extra sidecar container to be the client of the beaver server, but i dont like when mixed docker/local environment are to be setup for testing.

Comment (MaxHalford) [reply examples/taxis/README.md:31]: I get what you mean. How about using a virtual environment? A dedicated Docker instance could work too I guess. You decide! But in any case I would rather separate the server from the client.

Comment (dberardo-com) [reply examples/taxis/README.md:31]: alright, i will work that out in a future update, yesterday i just wanted to make a "checkpoint" having understood the basic principles of how Beaver is intended to work :D 

did i get the principle behind the "train.py" right? i mean is that an optional script to run? and would its execution be idempotent?

Comment (MaxHalford) [reply examples/taxis/README.md:31]: Sorry forgot to answer about that. Actually the train.py script was meant to be used when the only way to update a model was manually. Now there's an option to update it every time a label arrives. You can delete it :)

Comment (dberardo-com) [reply examples/taxis/README.md:31]: but again, just to undestand, will it be idempotent right? i mean in supervised learning the labels and features would be the same if train.py is executed twice. 
i guess that would not hold true for unsupervised learning (anomalies detection). but this is just a supposition here

Comment (MaxHalford) [reply examples/taxis/README.md:31]: Not it won't be. But you can re-run `upload_models.py` to reset the model

**Pull request closed without merging**